django/contrib/auth/__init__.py

```diff
@@ -37,6 +38,18 @@ def get_backends():
     return _get_backends(return_tuples=False)
 
 
+def csrf_is_enabled():
+    canonical_csrf_mw = import_string(CANONICAL_CSRF_MW)
+    for middleware_path in settings.MIDDLEWARE:
+        middleware = import_string(middleware_path)
+        if issubclass(middleware, canonical_csrf_mw):
+            return True
+    return False
+
+
+CSRF_IS_ENABLED = csrf_is_enabled()
```

Comment:
I'm not sure of understanding, do you mean to test specifically the skip of rotate_token when no CSRF is enabled? If so, I think it will be tested by the tests for no CSRF. Otherwise, can you elaborate more, please?
I remove it because test_complex_app (admin_scripts.tests.ManageCheck) failed with a circular import error - though not sure why - despite of all the rest of the test suite pass (the goal of the PR is to move token rotation to the middleware in the least obtrusive way).